indico/modules/bootstrap/controllers.py

```diff
@@ -0,0 +1,115 @@
+from markupsafe import Markup
+from requests.exceptions import HTTPError, Timeout
+
+from indico.core.config import Config
+from indico.core.db import db
+from indico.modules.auth import Identity, login_user
+from indico.modules.bootstrap.forms import BootstrapForm
+from indico.modules.cephalopod.util import register_instance
+from indico.modules.users import User
+from indico.util.i18n import _, get_all_locales, get_current_locale, parse_locale
+from indico.util.string import to_unicode
+from indico.web.flask.util import url_for
+
+import MaKaC
+from MaKaC.webinterface.rh.base import RH
+from MaKaC.common.info import HelperMaKaCInfo
+
+# TODO: set the time zone here once communities settings are available.
+
+
+class RHBootstrap(RH):
+
+    def _process_GET(self):
+        if db.session.query(db.session.query(User.id).exists()).one()[0]:
+            return redirect(url_for('misc.index'))
+        return render_template('bootstrap/bootstrap.html',
+                               selected_lang_name=parse_locale(get_current_locale()).language_name,
+                               language_options=sorted(get_all_locales().items(), key=itemgetter(1)),
+                               form=BootstrapForm(language=session.lang),
+                               timezone=Config.getInstance().getDefaultTimezone(),
+                               indico_version=MaKaC.__version__,
+                               python_version=python_version())
+
+    def _process_POST(self):
+        if db.session.query(db.session.query(User.id).exists()).one()[0]:
+            return redirect(url_for('misc.index'))
+        setup_form = BootstrapForm(request.form)
+        if not setup_form.validate():
+            flash(_("Some fields are invalid. Please, correct them and submit the form again."), 'error')
+            return redirect(url_for('bootstrap.index'))
+
+        # Creating new user
+        user = User()
+        user.first_name = to_unicode(setup_form.first_name.data)
+        user.last_name = to_unicode(setup_form.last_name.data)
+        user.affiliation = to_unicode(setup_form.affiliation.data)
+        user.email = to_unicode(setup_form.email.data)
+        user.is_admin = True
+
+        identity = Identity(provider='indico', identifier=setup_form.username.data, password=setup_form.username.data)
+        user.identities.add(identity)
+
+        db.session.add(user)
+        db.session.flush()
+
+        user.settings.set('timezone', Config.getInstance().getDefaultTimezone())
+        user.settings.set('lang', to_unicode(setup_form.language.data))
+
+        login_user(user, identity)
+        full_name = user.full_name  # needed after the session closes
+
+        transaction.commit()
+
+        # Configuring server's settings
+        minfo = HelperMaKaCInfo.getMaKaCInfoInstance()
+        minfo.setOrganisation(setup_form.affiliation.data)
+        minfo.setLang(setup_form.language.data)
+
+        flash(Markup(
```

Comment:
Not a big fan of it unless it's really tiny (such as just adding a small link). I'd rather go for `flash(Markup(render_template(...)))`